controllers/nodehealthcheck_controller.go

```diff
@@ -626,7 +624,8 @@ func (r *NodeHealthCheckReconciler) remediate(ctx context.Context, node *v1.Node
 	}
 
 	startedRemediation := resources.FindStatusRemediation(node, nhc, func(r *remediationv1alpha1.Remediation) bool {
-		return r.Resource.GroupVersionKind() == remediationCR.GroupVersionKind()
+		return r.Resource.GroupVersionKind() == remediationCR.GroupVersionKind() &&
```

Comment:
fixes general processing of CRs, didn't investigate further what potential this would have caused...